src/libstd/sys/windows/os.rs

```diff
@@ -192,6 +192,15 @@ impl<'a> Iterator for SplitPaths<'a> {
             Some(super::os2path(&in_progress))
         }
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        // There will be at most N + 1 entries, where N is the number of
+        // remaining semicolons.
+        let data = self.data.clone();
+        let semicolons = data.filter(|b| b == (';' as u16)).count();
```

Comment:
Hm, is it a good idea for the size_hint calculation to actually iterate over the data? Or am I misunderstanding what is happening here?

Reply:
This is definitely controversial / not obvious. My motivation to iterate over the data to calculate the size is because, in this case, we have a static guarantee that the thing being iterated over is in-memory, and that iterating over it will have no side-effects. Further, the iterator is over the number of entries in a directory, which is inherently bounded; and it's particularly common to collect the result of a dir entry iterator.